.github/workflows/java-pr.yml

```diff
@@ -40,7 +40,7 @@ on:
   workflow_dispatch:
 
 concurrency:
-  group: java-pr-${{ github.event.issue.number || github.run_id }}
+  group: ${{ github.workflow }}-${{ github.ref }}
```

Comment:
I don't think this will help much. `${{ github.workflow }}` will just resolve to java-pr anyways. `${{ github.ref }}` is probably better than `github.run_id` for scheduled runs (which only run every 12 hours anyways), but it is equivalent to `github.event.issue.number` for PRs since any PR runs will be the same PR number.

Note that jobs **are** getting cancelled today - https://github.com/GoogleCloudPlatform/DataflowTemplates/actions?query=is%3Acancelled

I'm fine taking this (its a bit cleaner) but I think the real problem is mostly just that our build times have gone up dramatically recently.